app/utils/watchdog.py

```diff
@@ -180,72 +257,74 @@ def on_deleted(self, event: FileSystemEvent) -> None:
                     "uuid": uuid,
                 }
             )
+        else:
+            # logger.debug(
+            #     "UUID resolution failed for mod deletion event: "
+            #     f"[event_scr_path_str: {event_scr_path_str}, uuid: {uuid}"
+            # )
+            return
 
     def on_modified(self, event: FileSystemEvent) -> None:
-        """
-        A callback function called when a file is modified.
+        """A function called when a file or directory is modified.
 
-        Parameters:
-        event: The event object representing the file modified event.
+        :param event: The event object representing the file or directory modification event.
+        :type event: FileSystemEvent
 
-        Returns: None
+        :return: None
         """
         event_scr_path_str = str(event.src_path)
-        # Resolve an existing UUID from our mapper
-        uuid = self.metadata_manager.mod_metadata_file_mapper.get(event_scr_path_str)
-
-        if not uuid:
-            logger.debug(f"No UUID found for modification event: {event_scr_path_str}")
+        # Explicitly check if the file created is an .acf file that we track metadata from
+        if self.__check_acf_file(event, Path(event_scr_path_str)):
             return
-
-        # Try to resolve a mod path from the from metadata
-        mod_path = self.metadata_manager.internal_local_metadata.get(uuid, {}).get(
-            "path"
-        )
-        # If we have a UUID and mod path resolved, proceed to update the mod
-        if not mod_path:
-            logger.debug(
-                f"UUID found for modification event, but no mod path. Event src_path: {event_scr_path_str} UUID: {uuid}"
+        # If we are still here, assume we need to try to resolve an existing UUID from our mod path -> UUID mapper
+        uuid = self.metadata_manager.mod_metadata_file_mapper.get(event_scr_path_str)
+        if uuid is not None:
+            # Try to resolve a mod path from the from metadata
+            mod_path = self.metadata_manager.internal_local_metadata.get(uuid, {}).get(
+                "path"
             )
-
-        logger.debug(f"Mod metadata modified: {event_scr_path_str}")
-        self.__cooldown_uuid_change(
-            callback={
-                "operation": "updated",
-                "path": mod_path,
-                "uuid": uuid,
-            }
-        )
+            # If we have a UUID and mod path resolved, proceed to update the mod
+            logger.debug(f"Mod metadata modified: {event_scr_path_str}")
+            self.__cooldown_uuid_change(
+                callback={
+                    "operation": "updated",
+                    "path": mod_path,
+                    "uuid": uuid,
+                }
+            )
+        else:
+            # logger.debug(
+            #     "UUID resolution failed for mod modified event: "
+            #     f"[event_scr_path_str: {event_scr_path_str}, uuid: {uuid}"
+            # )
+            return
```

Comment:
For cleanliness, I'd either just remove this, or uncomment out the logger

Reply:
@hk21702 Sounds good. It can easily be re-added.